gis/text_to_centreline/README.md

```diff
@@ -0,0 +1,800 @@
+- [How the Function Works](#How-the-Function-Works)
+  - [Step 1: Clean the data](#Step-1-Clean-the-data)
+  - [Step 2: Separate into different cases](#Step-2-Separate-into-different-cases)
+    - [2a) Entire Length](#2a-Entire-Length)
+    - [2b) Normal Cases - Two Intersections](#2b-Normal-Cases---Two-Intersections)
+    - [2c) Special Case 1 - An Intersection and An Offset](#2c-Special-Case-1---An-Intersection-and-An-Offset) 
+    - [2d) Special Case 2 - Two Intersections and At Least One Offset](#2d-Special-Case-2---Two-Intersections-and-At-Least-One-Offset)
+      - [The logic behind dealing with geom to be trimmed](#The-logic-behind-dealing-with-geom-to-be-trimmed)
+  - [Confidence output](#Confidence-output)
+- [Creating Bylaws Speed Limit Layer](#Creating-Bylaws-Speed-Limit-Layer)
+  - [Using the Function](#Using-the-Function)
+  - [Match to Centrelines and Categorize Bylaws](#Match-to-Centrelines-and-Categorize-Bylaws)
+  - [Final Clean Up](#Final-Clean-Up)
+- [Future Enhancement](#Future-Enhancement)
+  - [Wrapper function for `clean_bylaws_text`](#Wrapper-function-for-clean_bylaws_text)
+  - [Include `date_added` and `date_reapealed`](#Include-date_added-and-date_repealed)
+  - [Rename `highway` and `btwn`](#Rename-highway-and-btwn)
+- [Outstanding Work](#Outstanding-Work) 
+  - [Where did the bylaws fail](#where-did-the-bylaws-fail)
+  - [Tackle Cases with Known Geom Error](#tackle-cases-with-known-geom-error)
+  - [pgRouting returns the shortest path but street name different from `highway`](#pgrouting-returns-the-shortest-path-but-street-name-different-from-highway)
+  - [Direction stated on bylaws is not taken into account](#direction-stated-on-bylaws-is-not-taken-into-account)
+  - [Levenshtein distance can fail for streets that have E / W](#levenshtein-distance-can-fail-for-streets-that-have-e--w)
+  - [Include former municipality element of the "highway" field](#include-former-municipality-element-of-the-highway-field)
+  - [Tackle Cases with "an intersection and two offsets"](#tackle-cases-with-an-intersection-and-two-offsets)
+  - [Modify `con` (confidence level) definition to better reflect actual situation](#modify-con-confidence-level-definition-to-better-reflect-actual-situation)
+
+## Intro
+
+This is a `README` for the [`gis.text_to_centreline(bylaw_id, highway, fr, t)`](https://github.com/CityofToronto/bdit_data-sources/blob/text_to_centreline/gis/text_to_centreline/sql/function-text_to_centreline.sql) function, which is written in `postgresql`. The general purpose of this function is to take an input text description of a street location in the City of Toronto, and return an output of centreline segments that match this description. The input descriptions typically state the street that the bylaw is in effect on and the two intersections between which the bylaw. For example, you could use the function to get the centreline segments of Bloor Street between Royal York Road and St George Street.
+
```

Comment:
This section should include a brief explanation that the `posted_speed_limit_update` includes past work on transforming bylaws into centreline segments, including python code to transform the xml bylaws on Open Data into Postgresql to process.